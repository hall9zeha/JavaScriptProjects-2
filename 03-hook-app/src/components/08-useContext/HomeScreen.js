import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const HomeScreen = () => {
    //Buscará la instancia de UserContext en el árbol de componentes
    //ya que necesitamos acceder lo que hayamos compartido a través de ella
    const {user} = useContext(UserContext)
    console.log(user);
  return (
    <div>
        <h1>HomeScreen</h1>
        <pre>
          {JSON.stringify(user,null,3)}
        </pre>
        <hr/>
    </div>
  )
}
