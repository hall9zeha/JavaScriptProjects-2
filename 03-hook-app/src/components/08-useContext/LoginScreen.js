import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {
    const {user, setUser} = useContext(UserContext);

    const actionLogin = ()=>{
        const {id,name,email} = user;
        setUser({
            id,
            name,
            email,
            sigIn:true
        })
    }

  return (
    <div>
        <h1>LoginScreen</h1>
        <hr/>
        <button onClick={actionLogin} className='btn btn-primary '>Login</button>
    </div>
  )
}
