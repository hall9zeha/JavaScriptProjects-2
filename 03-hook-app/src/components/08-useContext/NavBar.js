import React from 'react'
import { Link, NavLink } from 'react-router-dom'

export const NavBar = () => {
  return (
        <nav className="navbar navbar-expand-sm  navbar-dark  bg-dark">
            <div className="container-fluid">
                <a className="navbar-brand" href="#">useContext</a>
              
                <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                <div className="navbar-nav">
                    {/* NavLink nos permite usar algunas clases de bootstrap */}
                    <NavLink exact activeClassName="active" className="nav-link nav-item " to='/'>Home</NavLink>
                    <NavLink exact activeClassName="active" className="nav-link nav-item" to='/about'>About</NavLink>
                    <NavLink exact activeClassName="active" className="nav-link nav-item" to='/login'>Login</NavLink>
                 
                </div>
                </div>
            </div>
        </nav>
   
  )
}
