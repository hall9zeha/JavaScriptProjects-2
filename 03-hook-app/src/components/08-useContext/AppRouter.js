import React from 'react'
import { BrowserRouter as  Router, Routes, Route } from 'react-router-dom'
import { AboutScreen } from './AboutScreen'
import { HomeScreen } from './HomeScreen'
import { LoginScreen } from './LoginScreen'
import { NavBar } from './NavBar'

export const AppRouter = () => {
  return (
    <Router>
        <div>
            <NavBar/>
            {/* El Routes de react-router es como un switch tradicional(Switch en react-router-dom está descontinuado), dependiendo de qué condición ejecutará tal o cual
            En este caso una dirección a un componente dentro de la ruta */}
            <Routes>
            {/* exact le dice a la ruta que debe buscar el path con todos sus caracteres y no por la primera coincidencia */}
                <Route exact path='/' Component={HomeScreen}/> 
                <Route exact path='/login' Component={LoginScreen}/>
                <Route exact path='/about' Component={AboutScreen}/>
                {/* Ponemos una ruta por defecto si no existiera ninguna de las anteriores */}
                <Route path='*' Component={HomeScreen}/> 
            </Routes>
        </div>
    </Router>
  )
}
