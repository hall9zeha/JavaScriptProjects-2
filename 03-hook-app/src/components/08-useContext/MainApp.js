import React, { useState } from 'react'
import { AppRouter } from './AppRouter'
import { UserContext } from './UserContext'

export const MainApp = () => {
    const userProvides = {
        id:123,
        name:"Martha",
        email:"martha@mail.com",
        sigIn:false
    }

    const [userState, setUserState] = useState(userProvides);

  return (
    //Envolver nuestros componentes con UseProvider nos permitir√° compartir 
    //elementos u objetos con sus componentes hijos 

    //Ahora enviaremos un objeto que contenga las propiedades de nuestro useState
    <UserContext.Provider value={
        {user:userState,
         setUser:setUserState
        }}>
        <AppRouter/>
    </UserContext.Provider>
  )
}

