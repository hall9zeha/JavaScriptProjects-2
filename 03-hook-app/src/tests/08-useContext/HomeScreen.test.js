import React from 'react';
import renderHook, {act, waitFor,render, screen} from '@testing-library/react';
import renderer from 'react-test-renderer';
import { HomeScreen } from '../../components/08-useContext/HomeScreen';
import { UserContext } from '../../components/08-useContext/UserContext';

describe('Test in <HomeScreen/> component', () => { 

    const userMock={
        id:1,
        name:'Barry',
        email:'barry@mail.com',
        sigIn:true
    }
    const wrapper = renderer.create(
        //Nuestro componente HomeScreen requiere un objeto de tipo "User"(o cualquiera que hayamos definido) que es proveido por 
        //UserContext así que le enviaremos un objeto también para la prueba
            <UserContext.Provider value={{user:userMock}}>
                <HomeScreen/>
            </UserContext.Provider>
    );

    test('should show successfully',()=>{
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

 })