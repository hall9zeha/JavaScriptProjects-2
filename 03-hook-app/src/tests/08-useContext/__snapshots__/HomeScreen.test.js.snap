// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <HomeScreen/> component should show successfully 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <pre>
    {
   "id": 1,
   "name": "Barry",
   "email": "barry@mail.com",
   "sigIn": true
}
  </pre>
  <hr />
</div>
`;
